### PR TITLE
compatibility with OTP 24.0-dev

### PR DESCRIPTION
**Pull request by tcrossland**

### CODE REVIEW

File: lib/mint/core/transport/ssl.ex [hunk 1]
```diff
@@ -623,23 +623,19 @@ defmodule Mint.Core.Transport.SSL do
...
-
-    defp get_valid_suites([{unquote(kex), unquote(cipher), unquote(mac)} | rest], valid) do
-      get_valid_suites(rest, valid)
+  def default_ciphers() do
+    if function_exported?(:ssl, :cipher_suites, 0) do
+      :ssl
+      |> apply(:cipher_suites, [])
+      |> Enum.reject(& &1 in @blacklisted_ciphers)
+    else
+      protocol_version = :tls_record.highest_protocol_version([])
```
Comment (ericmj) [inline lib/mint/core/transport/ssl.ex:632]: This is a private function, you may be able to use http://erlang.org/doc/man/ssl.html#versions-0 instead.

**Comment by ericmj** [comment]
This looks good. Can you look at the CI errors?

**Comment by tcrossland** [comment]
> This looks good. Can you look at the CI errors?

Sure, I'll try to sort this out sometime today.

File: lib/mint/core/transport/ssl.ex (hunk 1 above)
Comment (tcrossland) [reply lib/mint/core/transport/ssl.ex]: I've changed this to use the first version from `ssl_versions/0`. Does this seem reasonable?

**Comment by tcrossland** [comment]
I think this should be good to merge now

**Review by ericmj — APPROVED** [review]

**Comment by whatyouhide** [comment]
Thanks @tcrossland! 💟

**Comment by tcrossland** [comment]
> Thanks @tcrossland! 💟

Glad to contribute... I've seen there's a [test failure](https://github.com/elixir-mint/mint/runs/1911138478) when this was merged, but it seems unrelated to this change. What do you think?

**Comment by whatyouhide** [comment]
@tcrossland yes it's unrelated, no worries :)

**Comment by wojtekmach** [comment]
A new release with this patch would be appreciated :)

**Pull request merged**